### PR TITLE
User Stories 38 and 39

### PR DESCRIPTION
**Pull request by melatran**
-Completed user stories with Josh
-Admin can disable merchant
-When admin disables merchant, all the items from that merchant changes to inactive

-I did end up deleting this
`db/migrate/20200531225446_change_data_type_for_status.rb`
If it messes up on your end, let me know and I can put this file back into our migrations folder

### CODE REVIEW

File: app/controllers/admin/merchants_controller.rb [hunk 1]
```diff
@@ -1,6 +1,19 @@
...
\ No newline at end of file
+
+    def update
+      merchant = Merchant.find(params[:id])
+      merchant.update(status: 1)
+      merchant.inactivate_items
+
+      redirect_to "/admin/merchants"
+      flash[:notice] = "#{merchant.name} has been disabled"
+    end
```
Comment (Rostammahabadi) [inline app/controllers/admin/merchants_controller.rb:18]: Nice and clean method here

File: app/models/merchant.rb [hunk 2]
```diff
@@ -28,4 +28,9 @@ def distinct_cities
     item_orders.distinct.joins(:order).pluck(:city)
   end
 
+  def inactivate_items
+    items.each do |item|
+      item.update(active?: false)
+    end
+  end
```
Comment (Rostammahabadi) [inline app/models/merchant.rb:35]: Good use of a model method

File: app/controllers/sessions_controller.rb [hunk 3]
```diff
@@ -8,7 +8,7 @@ def new
 
   def create
     user = User.find_by(email: params[:email])
-    if user.authenticate(params[:password])
+    if user && user.authenticate(params[:password])
```
Comment (melatran) [inline app/controllers/sessions_controller.rb:11]: On my end, it fixed the authenticate nil class error.
Please let me know if it didn't work on your end

Comment (Rostammahabadi) [reply app/controllers/sessions_controller.rb:11]: That worked!

File: spec/features/admin/navigation_spec.rb [hunk 4]
```diff
@@ -60,5 +60,42 @@
             expect(page).to have_content(bike_shop.zip)
             expect(page).to have_link("View items sold by #{bike_shop.name}")
         end
-    end 
-end 
+
+        it "Admin can disable merchants and items become inactive" do
+
+          bike_shop = Merchant.create!(name: "Meg's Bike Shop", address: '123 Bike Rd.', city: 'Denver', state: 'CO', zip: 80203)
```
Comment (JPuchalla) [inline spec/features/admin/navigation_spec.rb:66]: Great testing.

File: app/models/merchant.rb (hunk 2 above)
Comment (Joshua-Tukman) [reply app/models/merchant.rb:35]: I still think that we will be able to get this done using straight Active Record

File: app/controllers/sessions_controller.rb (hunk 3 above)
Comment (Joshua-Tukman) [reply app/controllers/sessions_controller.rb:11]: Nice work!

**Pull request merged**